util/app/discovery/discovery.go

```diff
@@ -33,7 +33,7 @@ func Discover(ctx context.Context, repoPath string, enableGenerateManifests map[
 	apps := make(map[string]string)
 
 	// Check if it is CMP
-	conn, _, err := DetectConfigManagementPlugin(ctx, repoPath)
+	conn, _, err := DetectConfigManagementPlugin(ctx, repoPath, []string{})
```

Comment:
It's weird to me that this doesn't send any env. I think it's wrong. But I think that's a separate bug, not an issue with this PR.